### PR TITLE
Profile enhancements: hiking time (see 1912)

### PR DESCRIPTION
**Pull request by mariavkourouni**
[Demo](http://mf-geoadmin3.dev.bgdi.ch/kom_hikingTime/?dev3d=true)

### CODE REVIEW

**Comment by mariavkourouni** [comment]
Should I create a filter for the hiking time (hours - minutes) ?

**Comment by gjn** [comment]
Yes.

File: src/components/profile/ProfileService.js [hunk 1]
```diff
@@ -154,12 +176,57 @@ goog.require('ga_urlutils_service');
...
         };
 
+        //Distance
         this.distance = function(data) {
           if (data.length != 0) {
             return data[data.length - 1].dist;
           }
         };
 
+        //Hiking time
```
Comment (cedricmoullet) [inline src/components/profile/ProfileService.js]: Could you indicate the origin of this formula ?

File: src/components/profile/ProfileService.js [hunk 2]
```diff
@@ -8,6 +8,28 @@ goog.require('ga_urlutils_service');
     'pascalprecht.translate'
   ]);
 
+  module.filter('time', function() {
```
Comment (gjn) [inline src/components/profile/ProfileService.js]: Please prefix the filter with `ga`. e.g. `gaTimeFormat`

File: src/components/profile/style/profile.less [hunk 3]
```diff
@@ -1,3 +1,6 @@
+/*library added until we integrate the icons*/
+@import url("https://netdna.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css");
```
Comment (gjn) [inline src/components/profile/style/profile.less]: Please open an issue that we integrate the needed icon in our build and attribute the issue to @pfanguin or @pauloamado .

**Comment by gjn** [comment]
@mariavkourouni Added 2 comments. Please adress them.

File: src/components/profile/style/profile.less (hunk 3 above)
Comment (mariavkourouni) [inline src/components/profile/style/profile.less]: https://github.com/geoadmin/mf-geoadmin3/issues/2972

Comment (gjn) [inline src/components/profile/style/profile.less]: The icons are integrated in master. So you can remove this reference and rebase onto current master. @mariavkourouni

File: src/components/profile/ProfileService.js [hunk 4]
```diff
@@ -481,6 +566,9 @@ goog.require('ga_urlutils_service');
               this.elPoi[1]);
           this.updateElevationLabels('text.ga-profile-distance',
               this.dist);
+
+          this.group.select('text.ga-profile-hikTime')
+              .text(gaTimeFormatFilter(this.hikTime));
           };
```
Comment (oterral) [inline src/components/profile/ProfileService.js]: There is a weird indentation here

File: src/components/profile/ProfileService.js [hunk 5]
```diff
@@ -154,12 +177,59 @@ goog.require('ga_urlutils_service');
...
+                var dH = (data[i].alts[elevationModel] -
+                    data[i - 1].alts[elevationModel]) || 0;
+
+                //Constants of the formula
+                var arrConstants = [
+                  14.271, 3.6992, 2.5922, -1.4384,
+                   0.32105, 0.81542, -0.090261, -0.20757,
+                   0.010192, 0.028588, -0.00057466, -0.0021842,
+                   1.5176e-5, 8.6894e-5, -1.3584e-7, 1.4026e-6
+                    ];
```
Comment (oterral) [inline src/components/profile/ProfileService.js]: indentation

**Comment by ltclm** [comment]
not sure if the constants in the hiking formula are still correct, the wandern.ch excel has now other constants, updated 02.09.2016

**Comment by gjn** [comment]
I hear Limal singing: 'Neverending stoooooryyyyyyyyy, ahahahahah'

**Pull request merged**